plutarch-orphanage/src/PlutusLedgerApi/V2/Orphans.hs

```diff
@@ -0,0 +1,1271 @@
+              ( Either
+                  PLA.StakingCredential
+                  ( Either
+                      PLA.StakingCredential
+                      ( Either
+                          (PLA.StakingCredential, PLA.PubKeyHash)
+                          ( Either (PLA.PubKeyHash, PLA.PubKeyHash) (PLA.PubKeyHash, Integer)
+                          )
+                      )
+                  )
+              )
+          ) ->
+        PLA.DCert
+      outOf = \case
+        Nothing -> PLA.DCertGenesis
+        Just Nothing -> PLA.DCertMir
+        Just (Just (Left sc)) -> PLA.DCertDelegRegKey sc
+        Just (Just (Right (Left sc))) -> PLA.DCertDelegDeRegKey sc
+        Just (Just (Right (Right (Left (sc, pkh))))) -> PLA.DCertDelegDelegate sc pkh
+        Just (Just (Right (Right (Right (Left (pkh, pkh')))))) -> PLA.DCertPoolRegister pkh pkh'
+        Just (Just (Right (Right (Right (Right (pkh, e)))))) -> PLA.DCertPoolRetire pkh e
+
+-- | @since 1.0.0
+deriving via Integer instance Arbitrary PLA.Lovelace
+
+-- | @since 1.0.0
+deriving via Integer instance CoArbitrary PLA.Lovelace
+
+-- | @since 1.0.0
+instance Function PLA.Lovelace where
+  {-# INLINEABLE function #-}
+  function = functionMap coerce PLA.Lovelace
+
+{- | A 'CurrencySymbol' is either a BLAKE2b-244 hash or empty (representing the
+Ada symbol). In a fully-fair generator, this makes it vanishingly unlikely
+that the Ada symbol will be produced naturally (1 in 2^8^28 = 2^244) odds.
+QuickCheck doesn't give us the ability to represent these odds faithfully:
+thus, we merely make the Ada symbol as unlikely as we can. If you want to
+ensure that the Ada symbol is covered by your tests, you need to make
+dedicated tests for this. For this reason, we also don't shrink into the Ada
+symbol (indeed, we don't shrink at all).
+
+You may also want to look at 'NonAdaCurrencySymbol' if you're not interested
+in generating the Ada symbol in your case.
+
+@since 1.0.0
+-}
+instance Arbitrary PLA.CurrencySymbol where
+  {-# INLINEABLE arbitrary #-}
+  arbitrary =
+    PLA.CurrencySymbol . PlutusTx.toBuiltin @ByteString
+      <$> frequency
+        [ (1, pure "")
+        , (maxBound, unSizedByteString @28 <$> arbitrary)
+        ]
+
+-- | @since 1.0.0
+deriving via PlutusTx.BuiltinByteString instance CoArbitrary PLA.CurrencySymbol
+
+-- | @since 1.0.0
+instance Function PLA.CurrencySymbol where
+  {-# INLINEABLE function #-}
+  function = functionMap coerce PLA.CurrencySymbol
+
+{- | This instance can generate the Ada token name, with faithful odds. It is
+limited to generating printable ASCII names, rather than the full UTF-8
+range. See 'PlutusLedgerApi.QuickCheck.Utils.NonAdaTokenName' documentation
+for an explanation of why we chose to do this. We also shrink into the Ada
+token name: bear this in mind.
+
+If you don't want to generate (or shrink into) the Ada token name, consider
+using 'PlutusLedgerApi.QuickCheck.Utils.NonAdaTokenName' instead.
+
+@since 1.0.0
+-}
```

Comment:
Ok, so you're choosing to have ascii token names? if so that's perfectly fine, just want to make sure it's known that TokenNames are also plain bytes.

Reply:
I felt that for testing purposes, it's better to generate sensible(ish) names. UTF-8 with a bounded byte length is hard to generate, but printable ASCII is easy, so that's what I went with.